server/game/cards/15-W2D/WinnersCircleAuctionHouse.js

```diff
@@ -0,0 +1,37 @@
+const DeedCard = require('../../deedcard.js');
+
+class WinnersCircleAuctionHouse extends DeedCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: () => true,
+            match: this,
+            effect: ability.effects.dynamicControl(() => this.dudeCheck())
+        });
+    }
+
+    dudeCheck() {
+        let controllerDudeCount = 0;
+        let oppDudeCount = 0;
+
+        const gameLocation = this.getGameLocation();
+        if(!gameLocation) {
+            return 0;
+        }
+
+        const controllerDudes = gameLocation.getDudes().find(dude => dude.controller === this.controller);
+        if(controllerDudes) {
+            controllerDudeCount = 1;
+        }
+
+        const oppDudes = gameLocation.getDudes().find(dude => dude.controller === this.controller.getOpponent());
+        if(oppDudes) {
+            oppDudeCount = 1;
+        }
+
+        return controllerDudeCount + oppDudeCount;
```

Comment:
This can be done little bit shorter. I did not test nor format it so please check if it is ok.
```suggestion
        const gameLocation = this.getGameLocation();
        if(!gameLocation) {
            return 0;
        }

        return this.game.getPlayers().reduce((aggregate, player) => {
            if(gameLocation.getDudes().find(dude => dude.controller === player)) {
                return aggregate + 1;
            }
            return aggregate;
        }, 0);
```

Reply:
It works, just had to add a closing brace at the end.